pkg/http/core/kubernetes/deploy_test.go

```diff
@@ -117,6 +118,99 @@ var _ = Describe("Deploy", func() {
 		})
 	})
 
+	Context("generating the cluster", func() {
+		When("the kind is deployment", func() {
+			kind := "deployment"
+
+			BeforeEach(func() {
+				fakeKubeClient.ApplyWithNamespaceOverrideReturns(kubernetes.Metadata{Kind: kind}, nil)
+			})
+
+			It("sets the cluster", func() {
+				kr := fakeSQLClient.CreateKubernetesResourceArgsForCall(0)
+				Expect(kr.Cluster).To(Equal(fmt.Sprintf("%s %s", kind, "test-name")))
+			})
+		})
+
+		When("the kind is statefulSet", func() {
+			kind := "statefulSet"
+
+			BeforeEach(func() {
+				fakeKubeClient.ApplyWithNamespaceOverrideReturns(kubernetes.Metadata{Kind: kind}, nil)
+			})
+
+			It("sets the cluster", func() {
+				kr := fakeSQLClient.CreateKubernetesResourceArgsForCall(0)
+				Expect(kr.Cluster).To(Equal(fmt.Sprintf("%s %s", kind, "test-name")))
+			})
+		})
+
+		When("the kind is replicaSet", func() {
+			kind := "replicaSet"
+
+			BeforeEach(func() {
+				fakeKubeClient.ApplyWithNamespaceOverrideReturns(kubernetes.Metadata{Kind: kind}, nil)
+			})
+
+			It("sets the cluster", func() {
+				kr := fakeSQLClient.CreateKubernetesResourceArgsForCall(0)
+				Expect(kr.Cluster).To(Equal(fmt.Sprintf("%s %s", kind, "test-name")))
+			})
+		})
+
+		When("the kind is ingress", func() {
+			kind := "ingress"
+
+			BeforeEach(func() {
+				fakeKubeClient.ApplyWithNamespaceOverrideReturns(kubernetes.Metadata{Kind: kind}, nil)
+			})
+
+			It("sets the cluster", func() {
+				kr := fakeSQLClient.CreateKubernetesResourceArgsForCall(0)
+				Expect(kr.Cluster).To(Equal(fmt.Sprintf("%s %s", kind, "test-name")))
+			})
+		})
+
+		When("the kind is service", func() {
+			kind := "service"
+
+			BeforeEach(func() {
+				fakeKubeClient.ApplyWithNamespaceOverrideReturns(kubernetes.Metadata{Kind: kind}, nil)
+			})
+
+			It("sets the cluster", func() {
+				kr := fakeSQLClient.CreateKubernetesResourceArgsForCall(0)
+				Expect(kr.Cluster).To(Equal(fmt.Sprintf("%s %s", kind, "test-name")))
+			})
+		})
+
+		When("the kind is daemonSet", func() {
+			kind := "daemonSet"
+
+			BeforeEach(func() {
+				fakeKubeClient.ApplyWithNamespaceOverrideReturns(kubernetes.Metadata{Kind: kind}, nil)
+			})
+
+			It("sets the cluster", func() {
+				kr := fakeSQLClient.CreateKubernetesResourceArgsForCall(0)
+				Expect(kr.Cluster).To(Equal(fmt.Sprintf("%s %s", kind, "test-name")))
+			})
+		})
+
+		When("the kind is not a cluster type", func() {
+			kind := "pod"
```

Comment:
you don't need this variable just say `fakeKubeClient.ApplyWithNamespaceOverrideReturns(kubernetes.Metadata{Kind:"pod"}, nil)`